src/storage.js

```diff
@@ -15,16 +15,23 @@ class Storage {
         sessionstorage.setItem(key, obj);
     }
 
-    static getJSON(key: string): ?(Object|Array) {
+    static getJSON(key: string): any {
+        var item = this.getItem(key);
+
+        if (typeof item != "string") {
+            console.error("Cannot getJSON with non-string ");
+            return null;
+        }
+
         try {
-            return JSON.parse(this.getItem(key));
+            return JSON.parse(item);
         } catch (error) {
             console.error(error);
             return null;
         }
     }
 
-    static setJSON(key: string, obj: Object|Array): void {
+    static setJSON(key: string, obj: mixed): void {
```

Comment:
As discussed: If you can get it working I'll be super happy, but I've sunk a significant bit of time into it without any progress.